### PR TITLE
Handle seeing uncompressed sendgrid contact data

### PR DESCRIPTION
**Pull request by bleonard**
We are seeing connections failing because sendgrid is sending uncompressed data from their contacts export.

### CODE REVIEW

File: airbyte-integrations/connectors/source-sendgrid/source_sendgrid/streams.py [hunk 1]
```diff
@@ -299,7 +299,13 @@ def download_data(self, url: str, chunk_size: int = 1024) -> tuple[str, str]:
             tmp_file, "wb"
         ) as data_file:
             for chunk in response.iter_content(chunk_size=chunk_size):
-                data_file.write(decompressor.decompress(chunk))
+                try:
+                    # see if it's compressed. we are seeing some that are not all of a sudden.
+                    # but let's also guard against the case where sendgrid changes it back.
+                    data_file.write(decompressor.decompress(chunk))
+                except zlib.error as e:
+                    # it's not actually compressed!
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-sendgrid/source_sendgrid/streams.py:307]: Should we log a warning here to inform that an error has occurred?

**Review by lazebnyi — APPROVED** [review]
lgtm 

left a small question about logging and need bump connector version + add entry to changelog

File: airbyte-integrations/connectors/source-sendgrid/source_sendgrid/streams.py (hunk 1 above)
Comment (bleonard) [reply airbyte-integrations/connectors/source-sendgrid/source_sendgrid/streams.py:307]: For better or worse, this is the expected case I've seen at the moment. I would be curious how often it work and it doesn't work. What's the right way to log so I can see (datadog?) but not have it fill up the user-visible logs. Is that a thing?

**Comment by bleonard** [comment]
> need bump connector version + add entry to changelog

@lazebnyi did I do that right?

**Pull request merged**